api-2.2/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationRequestTranslatorImpl_2_2.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import static org.openmrs.module.fhir2.FhirConstants.OPENMRS_FHIR_EXT_MEDICATION_REQUEST_FULFILLER_STATUS;
+
+import javax.annotation.Nonnull;
+
+import org.hl7.fhir.r4.model.CodeType;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.MedicationRequest;
```

Comment:
Do we need to break this up somehow into r3 and r4 models?  Right now this translator will be used for both r3 and r4 requests.

Reply:
Right now, our R3 support is "whatever the built-in converters support". That is, our STU3 support requests all look (conceptually) like this:

```java
Resource resource = convertToR4(incomingResource);
Resource result = fhirService.performOperation(resource);
return convertToR3(result);
```

I _think_ extensions are properly handled by this (they've been around and are basically unchanged since R2, except for the addition of new `value` options).

This is by design so that we're not supporting an exponential number of classes.

Reply:
It would actually be fairly straight-forward to test if this gets translated correctly using the module integration tests.